src/ssvc/registry/base.py

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/env python
+"""
+Work in progress on an experimental registry object for SSVC.
+"""
+#  Copyright (c) 2025 Carnegie Mellon University.
+#  NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE
+#  ENGINEERING INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS.
+#  CARNEGIE MELLON UNIVERSITY MAKES NO WARRANTIES OF ANY KIND,
+#  EITHER EXPRESSED OR IMPLIED, AS TO ANY MATTER INCLUDING, BUT
+#  NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE OR
+#  MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE
+#  OF THE MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE
+#  ANY WARRANTY OF ANY KIND WITH RESPECT TO FREEDOM FROM
+#  PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
+#  Licensed under a MIT (SEI)-style license, please see LICENSE or contact
+#  permission@sei.cmu.edu for full terms.
+#  [DISTRIBUTION STATEMENT A] This material has been approved for
+#  public release and unlimited distribution. Please see Copyright notice
+#  for non-US Government use and distribution.
+#  This Software includes and/or makes use of Third-Party Software each
+#  subject to its own license.
+#  DM24-0278
+
+import logging
+from typing import Any, Literal, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+from ssvc._mixins import (
+    _Base,
+    _GenericSsvcObject,
+    _KeyedBaseModel,
+    _SchemaVersioned,
+    _Valued,
+)
+from ssvc.utils.field_specs import VersionString
+
+logger = logging.getLogger(__name__)
+
+SCHEMA_VERSION: str = "2.0.0"
+logger.debug(f"Using schema version {SCHEMA_VERSION} for SsvcObjectRegistry.")
+
+
+def lookup_type(module: str, type_name: str):
+    """
+    Lookup a type by its name in the specified module.
+    Args:
+        module: The module to search in.
+        type_name: The name of the type to lookup.
+
+    Returns:
+        The type if found, otherwise None.
+    """
+    import sys
+
+    mod = sys.modules.get(module)
+    if mod is None:
+        return None
+    return getattr(mod, type_name, None)
+
+
+def _get_obj_type(obj: object) -> str:
+    """
+    Get the type of the object for registry purposes.
+    Args:
+        obj: The object to check.
+
+    Returns:
+        str: If the object is of a recognized type, return the name of the recognized type.
+             Otherwise, return "other".
+    """
+    objtype = "other"  # default type if not recognized
+    recognized_types = [
+        lookup_type("ssvc.decision_points.base", "DecisionPoint"),
+        lookup_type("ssvc.decision_tables.base", "DecisionTable"),
+    ]
+    for t in recognized_types:
+        if t is not None and isinstance(obj, t):
+            objtype = t.__name__
+            break
+    return objtype
+
+
+class NonValuedVersion(BaseModel):
+    version: VersionString
+    obj: _GenericSsvcObject
+
+
+class ValuedVersion(BaseModel):
+    version: VersionString
+    obj: _GenericSsvcObject
```

Comment:
The `obj: _GenericSsvcObject` lines are a problem because it's not copying the full object, just the fields that are explicitly in the `_GenericSsvcObject` mixin. We need to resolve this before merging.